src/v/datalake/coordinator/iceberg_file_committer.cc

```diff
@@ -305,11 +307,17 @@ class table_commit_builder {
           "Adding {} files to Iceberg table {}",
           icb_files_.size(),
           table_id_);
+        // NOTE: a non-expiring tag is added to the new snapshot to ensure that
+        // snapshot expiration doesn't clear this snapshot and its commit
+        // metadata properties. The tag ensures that we retain them e.g. in
+        // case of external table updates and low snapshot retention.
         iceberg::transaction txn(std::move(table_));
         auto icb_append_res = co_await txn.merge_append(
           io,
           std::move(icb_files_),
-          {{commit_meta_prop, to_json_str(commit_meta)}});
+          {{commit_meta_prop, to_json_str(commit_meta)}},
+          commit_tag_name,
+          /*tag_expiration_ms=*/std::numeric_limits<long>::max());
```

Comment:
nit: this should be `std::numeric_limits<long long>::max()`. We never compile for a platform where this would result in 32 bit integer afaik but being explicit doesn't hurt.

long is defined as "at least" 32 bit by the C++ standard, on unix x86/arm 64bit it is actually a 64 bit number so all good, windows is 32 bit (who cares)

32 bit max as ms is ~24 days

Reply:
Good call, https://github.com/redpanda-data/redpanda/pull/25118